crates/loans/src/farming.rs

```diff
@@ -21,7 +21,7 @@ use sp_runtime::{traits::Zero, DispatchResult};
 use crate::*;
 
 impl<T: Config> Pallet<T> {
-    pub(crate) fn reward_account_id() -> Result<T::AccountId, DispatchError> {
+    pub fn reward_account_id() -> Result<T::AccountId, DispatchError> {
```

Comment:
Comparing how Substrate does this, I think we can get rid of the `Result` type here. Perhaps change this function to do something like this: https://github.com/paritytech/substrate/blob/e9b0facf70eeb08032cc7e83548c62f0b4a24bb1/frame/multisig/src/lib.rs#L529-L531

Reply:
Why is this blake hashing approach needed anyway, I don't think we use it elsewhere?

Reply:
I think the hashing is meant for generating entropy for the account.

Reply:
In the link that Greg posted it makes sense, since there they are generating a new account from input, and the input length is variable (possibly larger than the accountid). In our case, it's just a static accountid so I see no need to use this hashing approach. I'd argue that it's not adding entropy since the input is hard-coded. Yes, it'll make the bits look "more random", but at the end of the day it's still hardcoded. Having the bits look more random is not really providing any advantage. Note also that even parity is using accounts that are very much not random-looking, e.g. sovereign account look like `0x70617261f0070000000000000000000000000000000000000000000000000000`, which is just `"para" + parachainId`.

Reply:
switched to the infallible `into_sub_account_truncating` as discussed